com.btc.serviceidl/src/com/btc/serviceidl/generator/java/BasicJavaSourceGenerator.xtend

```
@@ -263,59 +259,63 @@ class BasicJavaSourceGenerator
-            return '''new «typeResolver.resolve("java.util.Vector")»<«IF is_failable»«typeResolver.resolve(JavaClassNames.COMPLETABLE_FUTURE)»<«ENDIF»«type»«IF is_failable»>«ENDIF»>()'''
-        }
-        else if (element instanceof StructDeclaration)
-        {
-            return '''new «typeResolver.resolve(element)»(«FOR member : element.allMembers SEPARATOR ", "»«IF member.optional»«typeResolver.resolve(JavaClassNames.OPTIONAL)».empty()«ELSE»«makeDefaultValue(member.type)»«ENDIF»«ENDFOR»)'''
-        }
-        else if (element instanceof EnumDeclaration)
-        {
-            return '''«toText(element)».«element.containedIdentifiers.head»''';
-        }
+        makeDefaultValue(element.type)
+    }
+
+    def public dispatch String makeDefaultValue(AbstractType element)
+    {
+        makeDefaultValue(element.referenceType ?: element.primitiveType ?: element.collectionType)
+    }
+
+    def public dispatch String makeDefaultValue(SequenceDeclaration element)
+    {
+        val type = toText(element.type)
+        val is_failable = element.failable
+
+        // TODO this should better use Collections.emptyList
+        '''new «typeResolver.resolve("java.util.Vector")»<«IF is_failable»«typeResolver.resolve(JavaClassNames.COMPLETABLE_FUTURE)»<«ENDIF»«type»«IF is_failable»>«ENDIF»>()'''
+    }
 
-        return '''0'''
+    def public dispatch String makeDefaultValue(StructDeclaration element)
+    {
+        '''new «typeResolver.resolve(element)»(«FOR member : element.allMembers SEPARATOR ", "»«IF member.optional»«typeResolver.resolve(JavaClassNames.OPTIONAL)».empty()«ELSE»«makeDefaultValue(member.type)»«ENDIF»«ENDFOR»)'''
+    }
+
+    def public dispatch String makeDefaultValue(EnumDeclaration element)
+    {
+        '''«toText(element)».«element.containedIdentifiers.head»'''
+    }
+
+    def public dispatch String makeDefaultValue(EObject element)
+    {
+        // TODO does this make sense? is this ever called?
+        '''0'''

```

Maybe throw an exception here to identify, which type is not covered by other dispatch methods? In future (e.g. if new types like DateTime or something like this) may be added, this can be useful to quickly figure out missing types.

---

That sounds better. I just kept the old behaviour, but I also found it quite strange. I will change it to throw an exception instead.